data_model/clusters/DiagnosticLogsCluster.xml

```diff
@@ -0,0 +1,136 @@
+are the sole property of their respective owners.
+
+This notice and disclaimer must be included on all copies of this document.
+
+Connectivity Standards Alliance
+508 Second Street, Suite 206
+Davis, CA 95616, USA
+-->
+<cluster xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="types types.xsd cluster cluster.xsd" id="0x0032" name="Diagnostic Logs" revision="1">
+  <revisionHistory>
+    <revision revision="1" summary="Initial Release"/>
+  </revisionHistory>
+  <classification hierarchy="base" role="utility" picsCode="DLOG" scope="Node"/>
+  <dataTypes>
+    <enum name="IntentEnum">
+      <item value="0" name="EndUserSupport" summary="Logs to be used for end-user support">
+        <mandatoryConform/>
+      </item>
+      <item value="1" name="NetworkDiag" summary="Logs to be used for network diagnostics">
+        <mandatoryConform/>
+      </item>
+      <item value="2" name="CrashLogs" summary="Obtain crash logs from the Node">
+        <mandatoryConform/>
+      </item>
+    </enum>
+    <enum name="StatusEnum">
+      <item value="0" name="Success" summary="Successful transfer of logs">
+        <mandatoryConform/>
+      </item>
+      <item value="1" name="Exhausted" summary="All logs has been transferred">
+        <mandatoryConform/>
+      </item>
+      <item value="2" name="NoLogs" summary="No logs of the requested type available">
+        <mandatoryConform/>
+      </item>
+      <item value="3" name="Busy" summary="Unable to handle request, retry later">
+        <mandatoryConform/>
+      </item>
+      <item value="4" name="Denied" summary="The request is denied, no logs being transferred">
+        <mandatoryConform/>
+      </item>
+    </enum>
+    <enum name="TransferProtocolEnum">
+      <item value="0" name="ResponsePayload" summary="Logs to be returned as a response">
+        <mandatoryConform/>
+      </item>
+      <item value="1" name="BDX" summary="Logs to be returned using BDX">
+        <mandatoryConform/>
+      </item>
+    </enum>
+  </dataTypes>
+  <commands>
+    <command id="0x00" name="RetrieveLogsRequest" response="RetrieveLogsResponse">
+      <access invokePrivilege="operate"/>
+      <mandatoryConform/>
+      <field id="0" name="Intent" type="IntentEnum">
+        <mandatoryConform/>
+      </field>
+      <field id="1" name="RequestedProtocol" type="TransferProtocolEnum">
+        <mandatoryConform/>
+      </field>
+      <field id="2" name="TransferFileDesignator" type="&lt;&lt;ref_DataTypeString&gt;&gt;">
+        <optionalConform/>
+        <constraint type="max" value="32"/>
+      </field>
+    </command>
+    <command id="0x01" name="RetrieveLogsResponse" direction="responseFromServer">
+      <access invokePrivilege="operate"/>
+      <mandatoryConform/>
+      <field id="0" name="Status" type="StatusEnum">
+        <mandatoryConform/>
+      </field>
+      <field id="1" name="LogContent" type="&lt;&lt;ref_DataTypeOctstr&gt;&gt;">
```

Comment:
ref_* are all over the place .....

Reply:
Same observation as above: we can fix the scraper, and regererate the XML afterwards.  Having the XML front and center means more eyes spotting bugs and hopefully more hands fixing bugs.  Also having this body of XML in the spec enables a parallel development of tooling to ensure alignment.

Reply:
https://github.com/csa-data-model/projects/issues/333 tracking the reference issues. Going to close this here to get the PR in, will update the XMLs once the scraper tooling is fixed.